libs/Microsoft.MixedReality.WebRTC.Native/include/interop_api.h

```diff
@@ -622,6 +630,21 @@ MRS_API mrsResult MRS_CALL mrsPeerConnectionRemoveLocalVideoTrack(
 MRS_API void MRS_CALL mrsPeerConnectionRemoveLocalAudioTrack(
     PeerConnectionHandle peerHandle) noexcept;
 
+MRS_API mrsResult MRS_CALL
+mrsAudioReadStreamCreate(PeerConnectionHandle peerHandle,
+                         int bufferMs,
```

Comment:
I'm not sure bufferMs should even be in the API. (I added it originally because I needed to test different values).

While on the topic of buffering, it's worth mentioning an issue if there's a hiccup (delay) in unity. There is no logic for 'catching up' and so in the worst case, and even with a great connection, the buffer can be completely full and all it's doing is adding latency. Half a second by default!

Reply:
> There is no logic for 'catching up'

This can be solved in `AudioReadStream::Read` by taking the newest n frames that fill the requested buffer, rather than taking the oldest ones, right? It seems that should be easy to do - or am i missing some fundamental issue?

Reply:
I think the issue is that if you always take the latest, then you run the risk of dropouts which is what the buffering is trying to avoid! So the idea is that you're willing to trade N ms of delay for reduced probability of dropouts.

You can do this by skipping some part of the buffer, but it's likely (?) that creates pops/artifacts. Maybe there's something better? Perhaps temporarily speed up the audio by messing with the sample rates, or something more complex which doesn't change the pitch.

Reply:
Yes, you're right (also if frames are pushed more frequently than when they are pulled that wont' work at all, but it shouldn't be our case).

Are we sure this is a problem though? I would expect OnAudioFilterRead to request more data after a hiccup to get up to speed.

In any case I have looked at how WebRTC handles audio packets buffering, doing it properly (and generically) is complicated unsurprisingly (see modules/audio_coding/net_eq/net_eq.cc). Though they have an `Accelerate` class that we might use directly (modules/audio_coding/net_eq/accelerate.h). Looks like some work so I'd be for logging this separately (if it is indeed an issue that can happen).

Reply:
I didn't look in to the pipeline of buffers, but would expect OnAudioFilterRead to be much more dumb/realtime and always ask for the same number of samples. I think it's only a short hop from there to handing to the output device.

Absolutely, it's complex and can be deferred. Perhaps there is some simple adaptive scheme we can use to choose the buffer size based on network conditions. Or perhaps webrtc is already doing all the smart stuff and we can reduce bufferMs to 20 or 30.

Reply:
I'll mark the API as experimental then and log an issue to investigate this more.

Reply:
> I didn't look in to the pipeline of buffers, but would expect OnAudioFilterRead to be much more dumb/realtime and always ask for the same number of samples.

That seems what's happening actually 😞